lib/alice/cache/cache.ex

```diff
@@ -210,6 +271,80 @@ defmodule Alice.Cache do
     #insert_many "presence_cache",    presences
   end
 
+  defp generic_cassandra_upsert(table, data) when is_binary(table) and is_map(data) do
+    # Alright.
+    # This is absolutely fucking stupid.
+    # NEVER do this for code that matters.
+    # Okay?
+    # Great. Don't judge me for this. It was the best way that
+    # my 2am-brain could figure out to handle this.
+    #
+    # Now that the disclaimer is out of the way...
+    # 
+    # Basically, the problem is how inserting works. This isn't like
+    # MongoDB, where we can just toss some JSON at it and pray. Instead,
+    # we have to know the keys - and their names! - in advance. This is
+    # an issue for the "naive" way of handling it.
+    # 
+    # So how do we solve this?
+    #
+    # Simple! Rather than guessing what keys will be present, we can 
+    # just look at what's present on the object, then use those keys 
+    # to build the query.
+    # 
+    # Discord *probably* isn't trying to CQL-inject us, so this is 
+    # relatively safe, for certain values of "this is absolutely fucking
+    # stupid and you should never do this in production code."
+    #
+    # Okay? Now don't judge me, I just wanted to get this done.
+    #
+    # I'm glad you understand.
+    #
+    # At least it's documented.
+    # 
+    # TODO: This might cause :fire: if ex. unexpected keys are present.
+    # Could possibly solve this by ex. grabbing all possible columns from 
+    # the table first, but that may have unexpected overhead(?)
+
+    try do
+      # Get all the keys
+      keys = Map.keys data
+      # Get the column string
+      cols = Enum.join keys, ", "
+      # Get the value string
+      vals = keys |> Enum.map(fn(x) -> ":" <> x end) |> Enum.join(", ")
+      # And now the insanity begins! Let's pray we're not about to get
+      # CQL-injected!
+      stmt = "INSERT INTO #{table} (#{cols}) VALUES (#{vals})"
+      # Then we need to make our mapping object
+      obj = keys |> Enum.reduce(%{}, fn(x, acc) -> 
+                Map.put(acc, x, data[x])
+              end)
+      Logger.debug "[CACHE] [DB] Running query: #{stmt} with data #{inspect obj, pretty: true}"
+      prep = Xandra.prepare! :cache, stmt, pool: DBConnection.Poolboy
```

Comment:
`Xandra.pretend! "that we're using prepared statements"`? 🤔

Reply:
@Shredder121 No, because it is using prepared statements. Just building them is, uh, less than sane.

Reply:
@Shredder121 No, because it is using prepared statements. Just building them is, uh, less than sane.